File: ReSharper.FSharp/src/Projects.FSharp/src/FSharpProjectLanguage.cs
using System;
using JetBrains.Annotations;
using JetBrains.ProjectModel;
using JetBrains.ProjectModel.Properties;

namespace JetBrains.Platform.ProjectModel.FSharp
{
  public class FSharpProjectLanguage : ProjectLanguage
  {
    public static readonly FSharpProjectLanguage Instance =
      new FSharpProjectLanguage("FSHARP", "F#", () => FSharpProjectFileType.Instance);

    public FSharpProjectLanguage([NotNull] string name, string presentableName,
      Func<ProjectFileType> projectFileType) : base(name, presentableName, projectFileType)
    {
    }
  }
}

File: ReSharper.FSharp/src/FSharp.Common/src/ProjectModel/FSharpProjectLanguage.fs
namespace JetBrains.ReSharper.Plugins.FSharp.ProjectModel

open JetBrains.ProjectModel.Properties
open JetBrains.ReSharper.Plugins.FSharp

type FSharpProjectLanguage(name, presentableName, projectFileType) =
    inherit ProjectLanguage(name, presentableName, projectFileType)

    static member val Instance =
        FSharpProjectLanguage("FSharp", "F#", fun () -> FSharpProjectFileType.Instance :> _) :> ProjectLanguage
